perceval/backend.py

```diff
@@ -533,6 +533,111 @@ def setup_cmd_parser(cls):
         raise NotImplementedError
 
 
+class BackendItemsGenerator:
+    def __init__(self, backend_class, backend_args, category, filter_classified=False,
+                 manager=None, archived_after=None, fetch_archive=False):
+        self.backend_class = backend_class
```

Comment:
We don't need to store most of these attributes. They'll be part of the backend. Most of the code of `init_backend` should be here.

Reply:
The iterator can also created here if we consider it makes sense.